Rubberduck.Parsing/SelectionExtensions.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Antlr4.Runtime;
+using Rubberduck.VBEditor;
+using System;
+
+namespace Rubberduck.Parsing
+{
+    /// <summary>
+    /// Provide extensions on selections & contexts/tokens
+    /// to assist in validating whether a selection contains
+    /// a given context or a token. 
+    /// </summary>
+    public static class SelectionExtensions
+    {
+        /// <summary>
+        /// Validates whether a token is contained within a given Selection
+        /// </summary>
+        /// <param name="selection">One-based selection, usually from CodePane.Selection</param>
+        /// <param name="token">An individual token within a module's parse tree</param>
+        /// <returns>Boolean with true indicating that token is within the selection</returns>
+        public static bool Contains(this Selection selection, IToken token)
+        {
+            return
+                (((selection.StartLine == token.Line) && (selection.StartColumn - 1) <= token.Column) 
+                    || (selection.StartLine < token.Line))
+             && (((selection.EndLine == token.EndLine()) && (selection.EndColumn - 1) >= (token.EndColumn())) 
+                    || (selection.EndLine > token.EndLine()));
+        }
+
+        /// <summary>
+        /// Validates whether a context is contained within a given Selection
+        /// </summary>
+        /// <param name="context">A context which contains several tokens within a module's parse tree</param>
+        /// <param name="selection">One-based selection, usually from CodePane.Selection</param>
+        /// <returns>Boolean with true indicating that context is within the selection</returns>
+        public static bool Contains(this ParserRuleContext context, Selection selection)
```

Comment:
I have been thinking about this function syntax. I had naively copied from the chat where it was first proposed. Now having implemented it in the test cases.... It looks weird to ask whether a `ParserContextRule` contains a `Selection`. That simply don't make sense. Can anyone else come up with a good reason why it should be extending the `ParserContextRule`, rather than `Selection`? If not, I am inclined to swap the parameters so that it extends the `Selection` to validate whether a `ParserRuleContext` is contained in the selection.

Reply:
I don't think this method is really needed; you can already do `context.GetSelection` and ask if that selection contains your other `Selection`, using the `Selection.Contains` member method.

Reply:
I think we should have extensions `Contains(ParserRuleContext context)` and `IsContainedIn(ParserRuleContext context)` using `context.GetSelection` and `Selection.Contains` for added simplicity of use.

Moreover, `context.GetSelection` needs a fix, because it does not take into accout that the last token might be a multi-line token. In other words, it has to be updated to use `EndLine` and 'EndColumn`.

Reply:
@MDoerner  just so I understand - you are proposed that `Selection` still should have 2 extensions methods using `ParserRuleContext` -- `Contains()` and `IsContainedIn`, correct?

I will make a note to fix the existing `GetSelection` method on the `ParserRuleContext` class and add to unit tests.